hand_eye_calibration_target_extractor/src/target_extractor.cc

```diff
@@ -131,6 +132,17 @@ int main(int argc, char** argv) {
       img.data = image_message->data;
       img.encoding = "mono16";
 
+      // scale up the image to take use of the full 16bit range before encoding
+      // it to 8bit range
+      cv_bridge::CvImagePtr cv_ptr_tmp;
+      cv_ptr_tmp =
+          cv_bridge::toCvCopy(img, sensor_msgs::image_encodings::MONO16);
+      double min, max;
+      cv::minMaxLoc(cv_ptr_tmp->image, &min, &max);
+      double scaling_factor = 65535 / max;
+      cv_ptr_tmp->image = scaling_factor * cv_ptr_tmp->image;
```

Comment:
I think there is a scaling function for images (which is probably be more efficient than this)

Reply:
Have a look at this: https://stackoverflow.com/questions/12023958/what-does-cvnormalize-src-dst-0-255-norm-minmax-cv-8uc1